### PR TITLE
Add missing checks to vdc, vdcuser and account templates

### PR DESCRIPTION
**Pull request by katia-e**
* Require state **install** to be OK in all actions accept install itself.
* Uninstall action always unsets state **install**.
* In `add_user` for `vdc` and `account` template add check that user was installed.
* Add argument validation in `add_user` actions: `name` should be given. 

closes: #74

### CODE REVIEW

File: templates/vdc/vdc.py [hunk 1]
```diff
@@ -146,15 +150,21 @@ def uninstall(self):
         '''
         Delete VDC
         '''
+        self.state.check('actions', 'install', 'ok')
```
Comment (chrisvdg) [inline templates/vdc/vdc.py]: This line would throw an exception if the install action state was not set to ok.
Should we do this in an uninstall/delete method or exit without error/exception and log that the install state was not ok?
@FastGeert?

**Comment by katia-e** [comment]
closes #77

File: templates/account/account.py [hunk 2]
```diff
@@ -102,9 +104,22 @@ def user_add(self, user):
         if not self.data['create']:
             raise RuntimeError('readonly account')
 
-        find = self.api.services.find(template_uid=self.VDCUSER_TEMPLATE, name=user['name'])
+        # check that username is given 
+        if not user.get('name'):
+            raise KeyError("failed to add user, field 'name' is required")
```
Comment (FastGeert) [inline templates/account/account.py]: ValueError here isof KeyError

File: templates/account/account.py [hunk 3]
```diff
@@ -102,9 +104,22 @@ def user_add(self, user):
...
+
+        find = self.api.services.find(template_uid=self.VDCUSER_TEMPLATE, name=service_name)
         if len(find) != 1:
-            raise ValueError('no account service found with name "%s"', user['name'])
+            raise ValueError('no vdcuser service found with name "%s"' % service_name)
+
+        # check that user was successfully installed
+        try:
+            find[0].state.check('actions', 'install', 'ok')
+        except StateCheckError:
```
Comment (FastGeert) [inline templates/account/account.py]: If the exception is not good enough, fix it where it gets thrown.

File: templates/vdc/vdc.py (hunk 1 above)
Comment (FastGeert) [reply templates/vdc/vdc.py]: Good catch @chrisvdg, when deleting, we should not care if the install succeeded.

File: templates/vdc/vdc.py [hunk 4]
```diff
@@ -237,10 +253,23 @@ def user_add(self, user):
         if not self.data['create']:
             raise RuntimeError('readonly cloudspace')
 
-        find = self.api.services.find(template_uid=self.VDCUSER_TEMPLATE, name=user['name'])
+        # check that username is given 
+        if not user.get('name'):
+            raise KeyError("failed to add user, field 'name' is required")
```
Comment (FastGeert) [inline templates/vdc/vdc.py]: ValueError for inpu validation

File: templates/vdc/vdc.py [hunk 5]
```diff
@@ -237,10 +253,23 @@ def user_add(self, user):
...
+        service_name = user['name'].split('@')[0]
+
+        find = self.api.services.find(template_uid=self.VDCUSER_TEMPLATE, name=service_name)
         if len(find) != 1:
             raise ValueError('no vdcuser service found with name "%s"', user['name'])
 
+        # check that user was successfully installed
+        try:
+            find[0].state.check('actions', 'install', 'ok')
+        except StateCheckError:
```
Comment (FastGeert) [inline templates/vdc/vdc.py]: Sam remark as previous. Should not catch and re-raise exceptions.

File: templates/vdcuser/vdcuser.py [hunk 6]
```diff
@@ -34,6 +34,16 @@ def get_fqid(self):
         return "%s@%s" % (self.name, provider) if provider else self.name
 
     def install(self):
+        '''
+        Install vdcuser
+        '''
+
+        try:
+            self.state.check('actions', 'install', 'ok')
```
Comment (FastGeert) [inline templates/vdcuser/vdcuser.py]: If would be nicer if we could do the following:
```python
if self.state.check('actions', 'install', 'ok', die=False):
    return
```

Comment (katia-e) [reply templates/vdcuser/vdcuser.py]: > `if self.state.get('actions','install') == {'install':'ok'}`

errors if `self.state == {}`

**Comment by katia-e** [comment]
closes #79

**Pull request merged**